Waterlife

okushama opened the pull request:
![2017-10-10_10 46 39](https://user-images.githubusercontent.com/717507/31383879-95e68ab2-adb4-11e7-84ed-492e3ef6943b.png)

Commit:
added piranha and sardine spawn eggs

Commit:
tweaked fleeing and following

also removed limit for monster eggs

tterrag1098 committed:
Fix seams on eagle ray render, clean up matrix ops

tterrag1098 committed:
Fix seams on eagle ray render, clean up matrix ops

okushama committed:
Merge pull request #155 from Tropicraft/whatelsebutfish

Whatelsebutfish

Commit:
added razz's dolphin

Commit:
sploosh

IAtlasFish that school together will only do so with the correct type

Commit:
animated dolphin

Commit:
smoothing those rotations!

also improved fish collision reaction

Commit:
support for varied textures

Cojomax99 committed:
Fix merge conflicts

Commit:
merge artefact removed

Commit:
oops?

Cojomax99 commented on src/main/java/net/tropicraft/core/client/entity/render/RenderDolphin.java:
```diff
@@ -0,0 +1,53 @@
+	@Override
+	public void doRender(EntityTropicraftWaterBase entityliving, double d, double d1, double d2, float f, float f1) {
+		if(refreshModel && entityliving.ticksExisted % 20 == 0) {
+			this.mainModel = new ModelDolphin();
```
Why do you use mainModel and dolphin, since dolphin is just equal to mainModel?

okushama replied:
it was on top of an old hack, is only debug code which isn't active right now

Cojomax99 commented on src/main/java/net/tropicraft/core/common/entity/underdasea/atlantoku/EntityTropicraftWaterBase.java:
```diff
@@ -222,9 +238,10 @@ public void onLivingUpdate() {
-						this.fleeEntity(closest);
-						this.isPanicking = true;
+					if(closest.isInWater()) {
+					//	this.fleeEntity(closest);
```
no more flee?

okushama replied:
fleeing the player was for debug purposes, i will be creating lists of types of entities to flee from, as well as lists for entities to snack on

Cojomax99 commented on src/main/java/net/tropicraft/core/common/entity/underdasea/atlantoku/EntityTropicraftWaterBase.java:
```diff
@@ -579,6 +595,21 @@ protected void resetHeight() {
+	
+	@Override
+	protected SoundEvent getAmbientSound() {
+		return SoundEvents.ENTITY_SQUID_AMBIENT;
```
We should find dolphin sounds eventually

okushama replied:
pinch your lips and pull air in

Cojomax99 commented on src/main/java/net/tropicraft/core/common/entity/underdasea/atlantoku/EntityTropicraftWaterBase.java:
```diff
@@ -614,4 +645,49 @@ public void setExpRate(int i) {
+	
+	public void assignRandomTexture() {
+		if(!world.isRemote) {
+			if(getTexturePool().length > 0) {
```
Since getTexturePool can return null, you should add a null check here.

okushama replied:
i could do, but for now it is only called if getTexturePool is not null

Cojomax99 commented on src/main/java/net/tropicraft/core/registry/EntityRenderRegistry.java:
```diff
@@ -92,12 +91,15 @@ public static void init() {
         registerEntityRender(EntityKoaHunter.class, new RenderKoaMan(Minecraft.getMinecraft().getRenderManager(), new ModelKoaMan(), 0.5F));
-
         //registerEntityRender(EntityWallStarfish.class, new RenderWallStarfish());
+        registerEntityRender(EntityLostMask.class, new RenderLostMask());
```
Oops, guess I messed up the placement of these two lines when merging.

Commit:
refactor

cleaned up for PR

Commit:
added missing eggs

fixed aqua lerping, was causing bad patting
added the beginning of ecosystem
fixed error because of left over render file

Cojomax99 commented on src/main/java/net/tropicraft/core/common/entity/underdasea/atlantoku/EntityTropicraftWaterBase.java:
```diff
@@ -99,6 +103,20 @@ public void entityInit() {
 	public void onLivingUpdate() {
 		super.onLivingUpdate();
+		//setDead();
+		if(this instanceof IPredatorDiet && world.getDifficulty().equals(EnumDifficulty.PEACEFUL)) {
```
Should this be && !world.getDifficulty().equals(EnumDifficulty.PEACEFUL) ?

okushama replied:
no

Cojomax99 reviewed (approved):
🐟 🍰 🐟
